official/vision/beta/projects/yolo/modeling/backbones/Darknet.py

```diff
@@ -0,0 +1,356 @@
+import tensorflow as tf
+import tensorflow.keras as ks
+import collections
+
+from official.vision.beta.modeling.backbones import factory
+from official.vision.beta.projects.yolo.modeling import building_blocks as nn_blocks
+
+
+# builder required classes
+class BlockConfig(object):
+
+  def __init__(self, layer, stack, reps, bottleneck, filters, kernel_size,
+               strides, padding, activation, route, output_name, is_output):
+    '''
+        get layer config to make code more readable
+
+        Args:
+            layer: string layer name
+            reps: integer for the number of times to repeat block
+            filters: integer for the filter for this layer, or the output depth
+            kernel_size: integer or none, if none, it implies that the the building block handles this automatically. not a layer input
+            downsample: boolean, to down sample the input width and height
+            output: boolean, true if the layer is required as an output
+        '''
+    self.layer = layer
+    self.stack = stack
+    self.repetitions = reps
+    self.bottleneck = bottleneck
+    self.filters = filters
+    self.kernel_size = kernel_size
+    self.strides = strides
+    self.padding = padding
+    self.activation = activation
+    self.route = route
+    self.output_name = output_name
+    self.is_output = is_output
+    return
+
+
+def build_block_specs(config):
+  specs = []
+  for layer in config:
+    specs.append(BlockConfig(*layer))
+  return specs
+
+
+def darkconv_config_todict(config, kwargs):
+  dictvals = {
+      "filters": config.filters,
+      "kernel_size": config.kernel_size,
+      "strides": config.strides,
+      "padding": config.padding
+  }
+  dictvals.update(kwargs)
+  return dictvals
+
+
+def darktiny_config_todict(config, kwargs):
+  dictvals = {"filters": config.filters, "strides": config.strides}
+  dictvals.update(kwargs)
+  return dictvals
+
+
+def maxpool_config_todict(config, kwargs):
```

Comment:
Can you add darktiny_config_todict and maxpool_config_todict as member functions for layer_registry?